scripts/cangen/cangen/main.py

```diff
@@ -0,0 +1,45 @@
+"""
+Author: Samuel Parent
+Date: 2024-04-13
+"""
+
+import argparse
+import logging
+
+from .generate_can_from_dbc import generate_can_from_dbc
+
+logging.basicConfig(level="INFO", format="%(levelname)-8s| (%(name)s) %(message)s")
+
+CONFIG_FILE_NAME = "config.yaml"
+
+
+def parse():
+    parser = argparse.ArgumentParser(description="DBC to C code generator")
+    parser.add_argument("project", type=str, help="Name of the project")
+    parser.add_argument(
+        "--log-level",
+        dest="level",
+        choices=["STATUS", "INFO", "VERBOSE", "DEBUG"],
+        default="INFO",
+        help="Log verbosity threshold",
+    )
+
+    # If parsing fails (ex incorrect or no arguments provided) then this exits with
+    # code 2.
+    return parser.parse_args()
```

Comment:
Since this program always requires CLI arguments, we can show the help screen if `cangen` is called with no arguments. Replace `return parser.parse_args()` with 

```python
return parser.parse_args(None if sys.argv[1:] else ["-h"])
```

You will need to import `sys` for this to work. Try to figure out what this expression evaluates to when different (or no) CLI args are passed, and why it shows the help screen when there aren't any CLI args